File: apps/zxtune-android/zxtune/src/main/java/app/zxtune/playback/stubs/CallbackStub.java
/**
 * @file
 * @brief
 * @author vitamin.caig@gmail.com
 */

package app.zxtune.playback.stubs;

import app.zxtune.TimeStamp;
import app.zxtune.playback.Callback;
import app.zxtune.playback.Item;
import app.zxtune.playback.PlaybackControl;

public abstract class CallbackStub implements Callback {

  @Override
  public void onInitialState(PlaybackControl.State state) {}

  @Override
  public void onStateChanged(PlaybackControl.State state, TimeStamp pos) {}

  @Override
  public void onItemChanged(Item item) {}

  @Override
  public void onError(String e) {}
}

File: apps/zxtune-android/zxtune/src/main/java/app/zxtune/playback/stubs/CallbackStub.kt
/**
 * @file
 * @brief
 * @author vitamin.caig@gmail.com
 */
package app.zxtune.playback.stubs

import app.zxtune.TimeStamp
import app.zxtune.playback.Callback
import app.zxtune.playback.Item
import app.zxtune.playback.PlaybackControl

abstract class CallbackStub : Callback {
    override fun onInitialState(state: PlaybackControl.State) {}
    override fun onStateChanged(state: PlaybackControl.State, pos: TimeStamp) {}
    override fun onItemChanged(item: Item) {}
    override fun onError(e: String) {}
}
